### PR TITLE
Add CI for unstable canary releases

### PR DESCRIPTION
**Pull request by chrisvxd**
This PR introduces 2 new types of release:

* **canary** - occurs on each master build. Tagged with `canary` on npm. Looks like `v0.7.1-canary.aa5d202`. This will enable us to test features merged to `main` before finalising.
* ~~**pr** - occurs on each PR build. Tagged with `pr` on npm. Looks like `v0.7.1-pr.aa5d202`. This will allow us to test features in PRs before finalising the PR. Useful for testing generator changes.~~

The semantic version is automatically calculated using the same tooling as the main release. No CHANGELOG is produced and currently no tag is produced either, otherwise each merge to master would result in a new tag.

### CODE REVIEW

File: .github/workflows/publish-canary.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    # Don't run on regular releases
+    if: "!startsWith(github.event.head_commit.message, 'release: ')"
+    steps:
+      - uses: actions/checkout@v2.0.0
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: "18.16.1"
```
Comment (monospaced) [inline .github/workflows/publish-canary.yml:19]: Not directly related to this PR, but notice we don't specify the node version anyway (in `.npmrc`; or in a `.node-version` or `package.json`) unless I missed it. I think we probably should do (one of those), to ensure that local dev builds are running with the same version as CI.

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -28,23 +28,8 @@ jobs:
...
-
-      - name: Publish plugin-heading-analyzer
-        run: cd packages/plugin-heading-analyzer && npm publish --access public
-        env:
-          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
-
-      - name: Publish create-puck-app
-        run: cd packages/create-puck-app && npm publish --access public
+      - name: Publish all packages
+        run: ./scripts/publish.sh pr
```
Comment (monospaced) [inline .github/workflows/publish.yml:32]: Should this be `pr`? For manual use?

**Review by monospaced — CHANGES_REQUESTED** [review]

File: .github/workflows/publish.yml (hunk 2 above)
Comment (chrisvxd) [reply .github/workflows/publish.yml:32]: Goooood catch!

Comment (chrisvxd) [reply .github/workflows/publish.yml:32]: Should definitely not be PR - this is the main release script

Comment (chrisvxd) [reply .github/workflows/publish.yml]: Fixed

File: package.json [hunk 3]
```diff
@@ -6,9 +6,13 @@
     "lint": "turbo run lint",
     "format:check": "prettier --check \"**/*.{ts,tsx,md}\"",
     "format": "prettier --write \"**/*.{ts,tsx,md}\"",
-    "release": "git fetch --tags && conventional-recommended-bump -p angular | xargs yarn version --no-git-tag-version --new-version$1 && yarn release-commit",
+    "release": "yarn release:prepare && yarn changelog && yarn release-commit",
+    "release:prepare": "git fetch --tags && conventional-recommended-bump -p angular  | xargs yarn version:auto $1",
```
[the remark this answers is not in the document]
Comment (chrisvxd) [reply package.json]: Fixed

File: package.json [hunk 4]
```diff
@@ -6,9 +6,13 @@
...
+    "release:canary": "yarn release:prepare && node scripts/get-unstable-version canary | xargs yarn version:auto $1",
+    "release:pr": "yarn release:prepare && node scripts/get-unstable-version pr | xargs yarn version:auto $1",
```
[the remark this answers is not in the document]
Comment (chrisvxd) [reply package.json]: Removed

File: .github/workflows/publish-canary.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (chrisvxd) [reply .github/workflows/publish-canary.yml:19]: We should probably specify the version in package.json, yeah. Will open a ticket.

**Review by monospaced — APPROVED** [review]
Guess we need to merge this to test it, but looks good now ✅

**Pull request merged**